src/js/msp/MSPHelper.js

```diff
@@ -796,6 +796,7 @@ MspHelper.prototype.process_data = function(dataHandler) {
                 console.log("Fw git rev:", FC.CONFIG.gitRevision);
 
                 if (semver.gte(FC.CONFIG.apiVersion, API_VERSION_1_46)) {
+                    FC.CONFIG.buildOptions = [];
```

Comment:
We have an INITIAL_CONFIG that I think does preciselly that:
https://github.com/betaflight/betaflight-configurator/blob/4d46da2a8b86eef13d565ece2638a82afdfd81e5/src/js/fc.js#L5-L52

It contains the `buildOptions`. Why it does not work?

Reply:
@McGiverGim

`this.CONFIG.buildOptions` is even retained when destroying the object and reassign with `INITIAL_CONFIG` instead of line 229. With this code `this.CONFIG.buildOptions` is not retaining previous values when switching FC.

```bash
this.CONFIG = null;
this.CONFIG = INITIAL_CONFIG;
```

Reply:
@McGiverGim doing a deep copy solves the issue.

https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy